test_queue_job_batch/tests/test_queue_job_batch.py

```diff
@@ -8,7 +8,7 @@ def test_batch(self):
         batch = self.env['queue.job.batch'].get_new_batch('TEST')
         self.assertFalse(batch.job_ids)
         model = self.env['test.queue.job'].with_context(
```

Comment:
Hey, I think we were going about this totally the wrong way, because we did not understand the [original PR](https://github.com/OCA/queue/pull/406/files).

Instead of using a text or JSON field, they're actually (JSON-)encoding the original context in a variable `keep_context` using the `JobEncoder`, which also encodes other details about the job.

Then instead of using `self.keep_context` as a boolean flag in the way we're doing now, they're using:

```
               if isinstance(self.keep_context, list):
                    context = {k: context.get(k) for k in self.keep_context}
```

So this then reinstates the context keys that were saved in that variable, one by one.

I guess if that works, then it avoids 1) Deviating from the other PR and 2) spending time getting our own JSON conversion right

The only comment on the current original PR was that by default it does not keep context, and most reviewers want it to keep context by default, and I agree with that too

Reply:
It already takes care of properly encoding and decoding Odoo recordsets:

https://github.com/OCA/queue/blob/11.0/queue_job/fields.py#L31